playbooks/installer/ospd/post.yml

```diff
@@ -19,30 +24,89 @@
 
       - block:
           - name: update file permissions
-            become: no
             file:
                 path: "{{ inventory_dir }}/id_rsa_overcloud"
                 mode: 0600
 
           - name: Update Inventory
-            become: no
             add_host:
                 name: "{{ item }}"
-                ansible_user: "heat-admin"
+                ansible_user: "{{ user }}"
             with_items: "{{ groups['overcloud_nodes'] }}"
         delegate_to: localhost
+        become: no
 
-      - name: Copy the keystonerc file for the tester
-        fetch:
-            src: "~/overcloudrc"
-            dest: "{{ inventory_dir }}/keystonerc"
-            flat: yes
+      - shell: "cat {{ undercloud_rc }} | grep AUTH | cut -d '=' -f 2"
+        register: oc_auth
 
-      - name: Copy the tempest-deployer-input.conf file for the tester
-        fetch:
-            src: "~/tempest-deployer-input.conf"
-            dest: "{{ inventory_dir }}/tempest-deployer-input.conf"
-            flat: yes
+      - shell: "cat {{ undercloud_rc }} | grep USERNAME | cut -d '=' -f 2"
+        register: oc_username
+
+      - shell: "cat {{ undercloud_rc }} | grep TENANT | cut -d '=' -f 2"
+        register: oc_tenant
+
+      - shell: "source {{ undercloud_rc }} && env | grep OS_PASSWORD | cut -d '=' -f 2"
+        args:
+            executable: /bin/bash
+        register: oc_password
+
+      - block:
+          - pip:
+                name: "{{ item.key }}"
+                version: "{{ item.value }}"
+                virtualenv: "{{ path_venv }}"
+            with_dict:
+                setuptools: "17.0"
+                pytz: "2016.4"
+                shade: "1.7.0"
+                pip: "8.1"
+
+          - name: update IP of overcloud nodes
+            vars:
+                ansible_python_interpreter: "/tmp/venv_server_facts/bin/python"
```

Comment:
IIRC, I've tried that, but the variable expansion is not applied happen here. I'll give it an another chance..